### DISCUSSION TITLE
SO fails to start after reboot

### DISCUSSION THREAD

**Discussion by justdiver** [Security-Onion-Solutions/securityonion#3167] [Unsupported Versions]
`****************************************************************************************************
* The following nodes in your Security Onion grid may need to be restarted due to package updates. * 
* If the node has already been patched, restarted and been up for less than 15 minutes, then it    *
* may not have updated it's restart_needed status yet. This will cause it to be listed below, even *
* if it has already been restarted. This feature will be improved in the future.                   *
****************************************************************************************************

    so2_standalone
    
Last login: Fri Feb 26 18:17:03 2021 from 10.50.50.73

sysadmins@so2:~$ 
sysadmins@so2:~$ sudo so-status
[sudo] password for sysadmins: 

Checking Docker status

    Docker -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    

Checking container statuses

    so-aptcacherng ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-cortex --------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-curator -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-dockerregistry --------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-elastalert ----------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-elasticsearch -------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-fleet ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-grafana -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-idstools ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-influxdb ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-kibana --------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-kratos --------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-mysql ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-nginx ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-playbook ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-redis ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-sensoroni ------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-soc ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-soctopus ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-steno ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-backend ------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-strelka-coordinator -------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-filestream --------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-frontend ----------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-gatekeeper --------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-manager ------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-suricata ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-telegraf ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-thehive -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-thehive-es ----------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-wazuh ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-zeek ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    

sysadmins@so2:~$ `

I'm not sure what would have caused this.

**Comment by weslambert** [comment]
Did these services eventually start, or is this still an issue?

**Reply by justdiver** [reply]
This is still an issue.

**Reply by justdiver** [reply]
Also, this was a long standing installation and not a new install.  Please let me know what additional information you might need.  I can post logs if necessary, I just don't know which logs would be the most helpful.

**Reply by cm-ops** [reply]
Is the salt-master service running?

**Comment by jdecker79** [comment]
I am new to SO and just setup an instance yesterday.  I am also experiencing this exact same thing.

**Reply by mgivens4182** [reply]
Fresh install ubuntu 18 tonight and a fresh install of SO same, will not start services.  Rebooted several times... No help.

**Reply by defensivedepth** [reply]
@mgivens4182 Please start a new Discussions post for your issues

**Comment by bouzeghoub** [comment]
so-start all

**Reply by rock-kobs** [reply]
simple command, works well, wonder if there's a way to fix it permanently. anyway thanks for the help.

**Comment by justdiver** [comment]
I ran soup on it.  It updated and started working.  I restarted the server and it's back into the same "wait-start" status as above.

**Reply by defensivedepth** [reply]
If you run `sudo salt-call state.highstate` do you see any errors?

**Reply by justdiver** [reply]
There were errors.  I will paste them below: 

----------
          ID: so-mysql
    Function: module.run
      Result: False
     Comment: 'so.mysql_conn': False
     Started: 18:29:35.099097
    Duration: 300620.799 ms
     Changes: 
----------
          ID: fleetdb
    Function: mysql_database.present
        Name: fleet
      Result: False
     Comment: One or more requisite failed: mysql.so-mysql
     Started: 18:36:31.821279
    Duration: 0.012 ms
     Changes:   
----------
          ID: fleetdbuser
    Function: mysql_user.present
      Result: False
     Comment: One or more requisite failed: fleet.fleetdb
     Started: 18:36:31.826315
    Duration: 0.01 ms
     Changes:   
----------
          ID: fleetdbpriv
    Function: mysql_grants.present
      Result: False
     Comment: One or more requisite failed: fleet.fleetdb
     Started: 18:36:31.830480
    Duration: 0.008 ms
     Changes:   
----------

          ID: create_playbookdbuser
    Function: mysql_user.present
        Name: playbookdbuser
      Result: False
     Comment: MySQL Error 1130: Host 'x.x.x.x' is not allowed to connect to this MySQL server
     Started: 18:36:57.668747
    Duration: 14.806 ms
     Changes:   
----------
          ID: query_playbookdbuser_grants
    Function: mysql_query.run
      Result: False
     Comment: MySQL Error 1130: Host 'x.x.x.x' is not allowed to connect to this MySQL server
     Started: 18:36:57.684082
    Duration: 6.367 ms
     Changes:   
----------
          ID: query_updatwebhooks
    Function: mysql_query.run
      Result: False
     Comment: MySQL Error 1130: Host 'x.x.x.x' is not allowed to connect to this MySQL server
     Started: 18:36:57.690896
    Duration: 4.984 ms
     Changes:   
----------
          ID: query_updatepluginurls
    Function: mysql_query.run
      Result: False
     Comment: MySQL Error 1130: Host 'x.x.x.x' is not allowed to connect to this MySQL server
     Started: 18:36:57.696484
    Duration: 6.875 ms
     Changes:   

In the above, x.x.x.x is the IP address of my server on which SO is installed.  It appears that MySQL has somehow disallowed itself from connecting?

**Reply by justdiver** [reply]
After running the salt-call command above, it shows the following: 

sysadmins@so2:~$ sudo so-status
[sudo] password for sysadmins: 

Checking Docker status

    Docker ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    

Checking container statuses

    so-aptcacherng --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-cortex -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-curator ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-dockerregistry ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-elastalert ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-elasticsearch ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-filebeat ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-fleet --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-grafana ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-idstools ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-influxdb ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-kibana -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-kratos -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-logstash ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-mysql --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-nginx --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-playbook ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-redis --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-sensoroni ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-soc ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-soctopus ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-steno --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-backend ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-coordinator ------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-filestream -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-frontend ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-gatekeeper -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-strelka-manager ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-suricata ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-telegraf ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-thehive ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-thehive-es ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-wazuh --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-zeek ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]

**Reply by defensivedepth** [reply]
Are you using a static IP or DHCP on your management interface? Did it recently change?

Do you have any other interfaces connected that have IPs on them?

**Reply by justdiver** [reply]
It's DHCP, but we have a reservation setup so it always gets the same address.  I'm not aware of it changing.  There are no other interfaces connected.  I rebooted it again this morning and it is still exhibiting the same behavior.

**Reply by defensivedepth** [reply]
It appears that something related to IP addressing has changed - howerver, you should be able to fix that Playbook/mysql error by running the following - `sudo so-playbook-reset`

Be aware that this will reset your Playbook instance and and changes you have made to Plays will be lost. It will also take a while to run.

After that is run, go ahead and run `sudo salt-call state.highstate` again and see if you are still seeing errors.

**Reply by justdiver** [reply]
I ran the playbook reset and then the salt-call.  After a reboot, I'm back in the "wait-start" status.  Is there a way to fix an IP addressing issue across all modules? 

(Thanks for your continued help on this).

**Reply by defensivedepth** [reply]
Run the following:

`sudo salt-call pillar.get global`

Do the following all have the same IP?

`fleet_ip`
`managerip`
`url_base`  --> this could be a dns name instead

Also - if you do a `sudo cat /etc/salt/minion`

does `mysql.host` also match that IP?

**Reply by justdiver** [reply]
They all report the same IP address and the URL resolves to that IP address as well.

**Reply by defensivedepth** [reply]
Ok, so everything works fine after you run (dont reboot):

`sudo salt-call state.highstate` ?

It only stops working after the reboot?

**Reply by justdiver** [reply]
Yes that is correct.

**Reply by defensivedepth** [reply]
Ok, go ahead and reboot, give it about 2 minutes and then run the following:

`systemctl status salt-*`

Sounds like the Salt service is not starting on reboot, so we need to confirm that.

Also check the salt logs for any errors:

`/opt/so/log/salt/master`
`/opt/so/log/salt/minion`

**Reply by justdiver** [reply]
sysadmins@so2:$ sudo systemctl status salt-*
[sudo] password for sysadmins: 
● salt-master.service - The Salt Master Server
   Loaded: loaded (/lib/systemd/system/salt-master.service; enabled; vendor preset: enabled)
   Active: active (running) since Thu 2021-03-18 15:22:45 UTC; 28min ago
     Docs: man:salt-master(1)
           file:///usr/share/doc/salt/html/contents.html
           https://docs.saltstack.com/en/latest/contents.html
 Main PID: 1925 (salt-master)
    Tasks: 43 (limit: 4915)
   CGroup: /system.slice/salt-master.service
           ├─1925 /usr/bin/python3 /usr/bin/salt-master
           ├─2231 /usr/bin/python3 /usr/bin/salt-master
           ├─2369 /usr/bin/python3 /usr/bin/salt-master
           ├─2375 /usr/bin/python3 /usr/bin/salt-master
           ├─2376 /usr/bin/python3 /usr/bin/salt-master
           ├─2379 /usr/bin/python3 /usr/bin/salt-master
           ├─2380 /usr/bin/python3 /usr/bin/salt-master
           ├─2381 /usr/bin/python3 /usr/bin/salt-master
           ├─2388 /usr/bin/python3 /usr/bin/salt-master
           ├─2389 /usr/bin/python3 /usr/bin/salt-master
           ├─2390 /usr/bin/python3 /usr/bin/salt-master
           ├─2391 /usr/bin/python3 /usr/bin/salt-master
           ├─2392 /usr/bin/python3 /usr/bin/salt-master
           └─2394 /usr/bin/python3 /usr/bin/salt-master

Mar 18 15:22:44 so2 systemd[1]: Starting The Salt Master Server...
Mar 18 15:22:45 so2 systemd[1]: Started The Salt Master Server.

● salt-minion.service - The Salt Minion
   Loaded: loaded (/lib/systemd/system/salt-minion.service; enabled; vendor preset: enabled)
   Active: active (running) since Thu 2021-03-18 15:22:46 UTC; 28min ago
     Docs: man:salt-minion(1)
           file:///usr/share/doc/salt/html/contents.html
           https://docs.saltstack.com/en/latest/contents.html
 Main PID: 2361 (salt-minion)
    Tasks: 7 (limit: 4915)
   CGroup: /system.slice/salt-minion.service
           ├─2361 /usr/bin/python3 /usr/bin/salt-minion
           ├─2609 /usr/bin/python3 /usr/bin/salt-minion
           ├─2635 /usr/bin/python3 /usr/bin/salt-minion
           └─2970 /usr/bin/python3 /usr/bin/salt-minion

Mar 18 15:50:13 so2 salt-minion[2361]: [INFO    ] Successfully queried Salt mine for the CA.
Mar 18 15:50:30 so2 salt-minion[2361]: [INFO    ] Running scheduled job: salt-minion-state-apply-test (runtime missed by 0.000468 seconds)
Mar 18 15:50:30 so2 salt-minion[2361]: [INFO    ] Loading fresh modules for state activity
Mar 18 15:50:31 so2 salt-minion[2361]: [INFO    ] Running state [/opt/so/log/salt/state-apply-test] at time 15:50:31.210036
Mar 18 15:50:31 so2 salt-minion[2361]: [INFO    ] Executing state file.touch for [/opt/so/log/salt/state-apply-test]
Mar 18 15:50:31 so2 salt-minion[2361]: [INFO    ] {'touched': '/opt/so/log/salt/state-apply-test'}
Mar 18 15:50:31 so2 salt-minion[2361]: [INFO    ] Completed state [/opt/so/log/salt/state-apply-test] at time 15:50:31.218338 (duration_in_ms=8.302)
Mar 18 15:50:31 so2 salt-minion[2361]: [INFO    ] Returning information for job: req
Mar 18 15:50:44 so2 salt-minion[2361]: [INFO    ] Runner completed: 20210318155043994063
Mar 18 15:50:44 so2 salt-minion[2361]: [INFO    ] Successfully queried Salt mine for the CA.
sysadmins@so2:$ sudo so-status

Checking Docker status

    Docker ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    

Checking container statuses

    so-aptcacherng --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ OK ]    
    so-cortex ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-curator ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-dockerregistry ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ OK ]    
    so-elastalert -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-elasticsearch ----------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-filebeat ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-fleet ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-grafana ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-idstools ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-influxdb ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-kibana ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-kratos ------------------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-logstash ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-mysql ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-nginx ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-playbook ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-redis ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-sensoroni --------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-soc --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-soctopus ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-steno ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-backend --------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-coordinator ----------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-filestream ------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-strelka-frontend -------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-strelka-gatekeeper ------------------------------------------------------------------------------------------------------------------------------------------------------------ [ WAIT_START ]    
    so-strelka-manager --------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-suricata ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-telegraf ---------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-thehive ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-thehive-es -------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-wazuh ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    
    so-zeek -------------------------------------------------------------------------------------------------------------------------------------------------------------------------- [ WAIT_START ]    

sysadmins@so2:$ tail /opt/so/log/salt/master
tail: cannot open '/opt/so/log/salt/master' for reading: Permission denied
sysadmins@so2:$ sudo !!
sudo tail /opt/so/log/salt/master
2021-03-18 15:19:36,976 [salt.utils.parsers:1104][WARNING ][1847] Master received a SIGTERM. Exiting.
sysadmins@so2:$ sudo tail /opt/so/log/salt/minion
2021-03-18 15:50:30,645 [salt.state       :1144][INFO    ][15886] Loading fresh modules for state activity
2021-03-18 15:50:31,210 [salt.state       :2039][INFO    ][15886] Running state [/opt/so/log/salt/state-apply-test] at time 15:50:31.210036
2021-03-18 15:50:31,211 [salt.state       :2071][INFO    ][15886] Executing state file.touch for [/opt/so/log/salt/state-apply-test]
2021-03-18 15:50:31,218 [salt.state       :319 ][INFO    ][15886] {'touched': '/opt/so/log/salt/state-apply-test'}
2021-03-18 15:50:31,218 [salt.state       :2226][INFO    ][15886] Completed state [/opt/so/log/salt/state-apply-test] at time 15:50:31.218338 (duration_in_ms=8.302)
2021-03-18 15:50:31,224 [salt.minion      :2150][INFO    ][2609] Returning information for job: req
2021-03-18 15:50:44,548 [salt.client.mixins:441 ][INFO    ][2970] Runner completed: 20210318155043994063
2021-03-18 15:50:44,549 [salt.loaded.ext.engines.checkmine:15  ][INFO    ][2970] Successfully queried Salt mine for the CA.
2021-03-18 15:51:15,625 [salt.client.mixins:441 ][INFO    ][2970] Runner completed: 20210318155115075451
2021-03-18 15:51:15,626 [salt.loaded.ext.engines.checkmine:15  ][INFO    ][2970] Successfully queried Salt mine for the CA.
sysadmins@so2:$

**Reply by justdiver** [reply]
The minion log has a lot of errors, but it's pretty long.  18k lines.  Not sure what I'm looking at here.  Should I post it?

**Reply by defensivedepth** [reply]
@justdiver is this still an issue?

**Reply by justdiver** [reply]
Yes @defensivedepth

**Reply by defensivedepth** [reply]
@justdiver Please update to the newly released .50 - I believe this will help.

**Comment by ghost** [comment]
Not sure if it is wholly relevant to the issues others are having, but none of my containers would start either. The problem ended up being a syntax error under thresholding in the global pillar (/opt/so/saltstack/local/pillar/global.sls). I had an extra space on one of the SIDS That kept the pillar from compiling / rendering. 

Maybe not THE answer, but certainly something to double check if you've done any mucking about in there.

Happy Hunting!

**Reply by petiepooo** [reply]
@jcail how did you detect that error; did it show up in a logfile that you noticed?

**Reply by ghost** [reply]
It showed up indirectly when I tried restarting or refreshing docker. Red letters that said failed to compile. In the case of soup, there was a reference to thresholding in the soup output & log. Those two items combined led me back to just reviewing the thresholding section, which is the only place I had made any changes. Here is what caused the whole thing to come to a screeching halt (including NGINX - as in no web console access). It is difficult to see, but in the CORRECT one, there are 4 spaces before the rule.uuid. On the INCORRECT one, there are 5. As soon as I corrected that, all the containers went to starting and in a couple of minutes, I was back into the console.

CORRECT
sids:
    2101411:
    - suppress:
        gen_id: 1
        track: by_src
        ip:

INCORRECT
sids:
     2101411:
    - suppress:
        gen_id: 1
        track: by_src
        ip:

**Reply by petiepooo** [reply]
Line indents did not show up in your paste, but I get the gist.  I didn't realize json was so picky in line formatting.  Thanks for letting me know.
